### PR TITLE
bvt retry on api timeouts

### PR DESCRIPTION
**Pull request by niranjan92**
https://github.com/Shippable/pm/issues/8506

### CODE REVIEW

File: _common/shippable/Adapter.js [hunk 1]
```diff
@@ -1332,7 +1332,8 @@ function _performCall(bag, next) {
           err = err || res.statusCode;
 
         if ((res && res.statusCode > 299) || err) {
-          if ((res && res.statusCode >= 500) || connectionError) {
+          if ((res && res.statusCode >= 500) || connectionError ||
```
Comment (shrivara) [inline _common/shippable/Adapter.js]: Lets clean up the conditions here: `connectionError || res.statusCode >= 500 || res.statusCode === 408`.

Comment (shrivara) [inline _common/shippable/Adapter.js]: We need to clean this whole method.
For us, an error is when there is an error, or statusCode >= 500 or statusCode is 408.

So, if there is no error, check for these errors and set the err object to this. Get rid of everything else like connectionError.

Comment (niranjan92) [inline _common/shippable/Adapter.js]: if rc is down, we get connectionError. I think we should also handle that too.

**Comment by niranjan92** [comment]
closing as this is not needed

**Pull request closed without merging**